### PR TITLE
[AArch64] Refactor ALUOp3

### PR DESCRIPTION
**Pull request by sparker-arm**
As well as adding generic pattern for msub along with runtests
for madd and msub.

Copyright (c) 2022, Arm Limited.

### CODE REVIEW

File: cranelift/codegen/src/isa/aarch64/lower.isle
Comment (cfallin) [inline cranelift/codegen/src/isa/aarch64/lower.isle:74]: The `def_inst` here is now optional, given the implicit conversions we recently added; so this pattern can be written as `(isub x (imul y z))`, which is a bit easier to read!

**Review by cfallin — APPROVED** [review]
LGTM modulo some suggestions below, thanks!

**Review by cfallin — APPROVED** [review]
Updates LGTM, thanks! I'm going to wait to merge this until after #3989 (which I plan to do in the morning), as that PR and this one will likely have merge conflicts. I'm happy to do the work to resolve the conflicts, though, and will do so once the former merges.

**Comment by sparker-arm** [comment]
Okay, thanks.

**Pull request merged**
